specs/beacon-chain.md

```diff
@@ -870,6 +905,81 @@ while len(crystallized_state.persistent_committee_reassignments) > 0 and crystal
     crystallized_state.persistent_committees[rec.shard].append(rec.validator_index)
 ```
 
+## Shard block processing
+
+A block on a shard can be processed only if its `parent` has been accepted. To validate a block header on shard `shard_id`, compute as follows:
+
+* Verify that `beacon_chain_ref` is the hash of the `slot`'th block in the beacon chain.
+* Let `state` be the state of the beacon chain block referred to by `beacon_chain_ref`. Let `validators` be `[validators[i] for i in state.current_persistent_committees[shard_id]]`.
+* Assert `len(attester_bitfield) == ceil_div8(len(validators))`
+* Let `curblock_proposer_index = hash(state.randao_mix + bytes8(shard_id)) % len(validators)`. Let `parent_proposer_index` be the same value calculated for the parent block.
+* Make sure that the `parent_proposer_index`'th bit in the `attester_bitfield` is set to 1.
+* Generate the group public key by adding the public keys of all the validators for whom the corresponding position in the bitfield is set to 1. Verify the `aggregate_sig` using this as the pubkey and the `parent_hash` as the message.
+
+Note that at network layer we expect blocks to be broadcasted along with the signature from the `curblock_proposer_index`'th validator in the validator set for that block.
+
+### Verifying shard block data
+
+At network layer, we expect a shard block header to be broadcasted along with its `block_body`. First, we define a helper function that takes as input beacon chain state and outputs the max block size in bytes:
+
+```python
+def calc_block_maxbytes(state):
+    max_grains = MAX_SHARD_BLOCK_SIZE // CHUNK_SIZE
+    validators_at_max_committees = SHARD_COUNT * TARGET_COMMITTEE_SIZE
+    grains = min(len(get_active_validator_indices(state.validators)) * max_grains // validators_at_max_committees,
```

Comment:
This could use a clarifying comment. Something like the following
```
# number of grains per block is proportional to the number of validators
# up until `validators_at_target_committee_size`
```